src/app/main/main.component.html

```diff
@@ -120,9 +120,29 @@
                   <app-rule-input [prop]="prop" [columns]="columnNames" [stats]="getPropStats(prop.name)" [formGroup]="getPropFormGroup(prop.name)"></app-rule-input>
                 </mat-expansion-panel>
               </mat-accordion>
-              <button mat-raised-button color="primary" (click)="updateStyles()">Apply Style</button>
+              <button mat-raised-button color="primary" (click)="onApplyStylesClicked()" class="apply-style-button" >Apply Style</button>
             </form>
           </mat-step>
+          <mat-step [stepControl]="sharingFormGroup" label="Share">
+	    <form [formGroup]="sharingFormGroup">
+	      <mat-form-field class="sql-form-field">
+		<p class="sql-caption"> <b>Please read before creating a link:</b> </p>
+		<ul class="sql-caption">
+		  <li> Creating a sharing link will save the project ID, BigQuery Job ID, and your styling rules (GeoViz does not store the SQL text or the underlying data from BigQuery). </li>
+		  <br>
+                  <li> Treat the link like a password:
+		    <ul>
+		    <br>
+		      <li> Anyone with the link can restore the styling rules. </li>
+		      <br>
+		      <li> The SQL for the BigQuery job is only available to those with sufficient access level to the GCP project that was used to run the job. </li>
```

Comment:
Suggested text updates:

> **IMPORTANT:** Creating a sharing link will save information about the query and style settings. Any user with the link can restore these settings. However, results returned by the query _will not_ be stored, and the the ability to execute the query and view the results is restricted to users with the necessary permissions on the selected Google Cloud Platform project.

Rationale:

1. While we don't store the SQL text, we do store a hash, and that distinction is subtle enough that I think we might as well not claim we aren't storing it.
2. Simpler and more future-proof to say "query settings" than to enumerate exactly which are stored.
3. The "like a password" metaphor is probably too strong. :)

Reply:
I like your text however I do want to say that we don't store anything about the SQL text (we don't even store a hash). Are you saying that the encrypted hash of the styles is still part of SQL text because it may have column names?

Reply:
Between the facts that we ...

1. do appear to use the SQL for the encryption key; changing the query will change what we store
2. do auto-magically restore the query when the user loads the saved URL

... I think the fact that the SQL was not technically stored _in Firebase_ is, while good, not going to mean anything to 95% of users. And I worry that the fact that the SQL does get restored (just not from Firebase) introduces confusion that wouldn't have been caused if we didn't promise we weren't storing your SQL.

Reply:
Sounds good.

Reply:
SG